src/main/java/controller/LottoController.java

```diff
@@ -0,0 +1,96 @@
+package controller;
+
+import domain.*;
+import view.InputView;
+import view.OutputView;
+import java.util.List;
+
+public class LottoController {
+    private final InputView inputView;
+    private final OutputView outputView;
+    private final LottoService lottoService;
+
+    public LottoController() {
+        this.inputView = new InputView();
+        this.outputView = new OutputView();
+        this.lottoService = new LottoService();
+    }
+
+    public void run() {
+        try {
+            Money purchaseAmount = inputPurchaseAmount();
+
+            Purchase purchase = inputPurchase(purchaseAmount);
+
+            LottoTickets tickets = lottoService.purchaseLottos(purchase);
+            outputView.printPurchaseResult(tickets);
+
+            WinningLotto winningLotto = inputWinningNumbers();
+
+            LottoResult result = lottoService.calculateResult(tickets, winningLotto);
+            outputView.printResult(result, purchaseAmount);
+
+        } catch (IllegalArgumentException e) {
+            outputView.printError(e.getMessage());
+        } catch (Exception e) {
+            outputView.printError("예상치 못한 오류가 발생했습니다.");
+        }
+    }
+
+    private Money inputPurchaseAmount() {
+        while (true) {
+            Money money = tryInputMoney();
+            if (money != null) {
+                return money;
+            }
+        }
+    }
+
+    private Money tryInputMoney() {
+        try {
+            int amount = inputView.inputPurchaseAmount();
+            return new Money(amount);
+        } catch (IllegalArgumentException exception) {
+            outputView.printError(exception.getMessage());
+            return null;
+        }
+    }
```

Comment:
이러니까 indent가 또 2가 되죠? 살짝 팁을 드려보자면

```suggestion
    private Money inputPurchaseAmount() {
        try {
            int amount = inputView.inputPurchaseAmount();
            return new Money(amount);
        } catch (IllegalArgumentException exception) {
            outputView.printError(exception.getMessage());
            return inputPurchaseAmount();
        }
    }
```

이렇게 해볼 수 있지 않을까요?